akka-distributed-data/src/main/scala/akka/cluster/ddata/ORSet.scala

```diff
@@ -307,6 +333,46 @@ final class ORSet[A] private[akka] (
     }
   }
 
+  private def mergeAddDelta(that: ORSet[A]): ORSet[A] = {
```

Comment:
My previous implementation uses a hack - it treats full merge as a black-box, it re-applies the delta on original `ORSet` to use full merge on those two. Your way is better.